lib/Sema/CSDiagnostics.cpp

```diff
@@ -5338,6 +5338,8 @@ bool MissingArgumentsFailure::diagnoseSingleMissingArgument() const {
     // fn(argX, argY):
     //   fn(argX, argY[, argMissing])
     if (args->empty()) {
+      if (!args->getRParenLoc().isValid())
+        return false;  
```

Comment:
I think this is too drastic. If there are no parens we can still produce a diagnostic and suggest a fix-it. The diagnostic can be attached to the end of the "fn" - `fnExpr->getEndLoc()` and suggest adding parens.